File: problems/005/005.js
function gcd(a, b) {
    if (a < b) {
        var t = a;
        a = b;
        b = t;
    }

    var r;
    while (b !== 0) {
        r = a % b;
        a = b;
        b = r;
    }

    return a;
}

function lcm(a, b) {
    return a * b / gcd(a, b);
}

function f() {
    var n = 2;

    var i;
    for (i = 3; i < 21; i++) {
        n = lcm(n, i);
    }

    return n;
}

function toNs(ts) {
    return ts.sec * 1000000000 + ts.nsec;
}

var clock = require('posix-clock');

if (process.argv.length === 3) {
    var i;
    var iters = parseInt(process.argv[2], 10);

    var start = clock.gettime(clock.MONOTONIC);
    for (i = 0; i < iters; i++) {
        f();
    }
    var end = clock.gettime(clock.MONOTONIC);

    console.log(toNs(end) - toNs(start));
} else {
    console.log(f());
}

File: problems/005/005.lua
posix = require "posix"

local function gcd(a, b)
    if a < b then
        local t = a
        a = b
        b = t
    end

    while b ~= 0 do
        a, b = b, a % b
    end

    return a
end

local function lcm(a, b)
    return a * b / gcd(a, b)
end

local function f()
    local n = 2

    for i = 3,20 do
        n = lcm(n, i)
    end

    return n
end

function toNs(ts)
    return ts[1] * 1000000000 + ts[2]
end

if #arg == 0 then
    print(f())
elseif #arg == 1 then
    iters = tonumber(arg[1])
    start = { posix.clock_gettime("monotonic") }
    for i = 1,iters do
        f()
    end
    end_ = { posix.clock_gettime("monotonic") }
    print(toNs(end_) - toNs(start))
end
